src/components/TodoItem.tsx

```diff
@@ -0,0 +1,75 @@
+import { useContext, useRef, useState } from 'react';
+import { Todo } from '../types/Todo';
+// eslint-disable-next-line import/no-cycle
+import { TodoContext } from './TodosContext';
+
+interface Props {
+  todo: Todo;
+  toggleCompleted: (id: number) => void;
+  deleteTodo: (id: number) => void;
+}
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const todoContext = useContext(TodoContext);
+  const [isEditing, setEditing] = useState(false);
+  // const [title, setTitle] = useState(todo.title);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleToggleTodo = () => {
+    todoContext?.toggleCompleted(todo.id);
+  };
+
+  const handleDeleteTodo = () => {
+    todoContext?.deleteTodo(todo.id);
+  };
+
+  return (
+    <div
+      data-cy="Todo"
+      className={todo.completed ? 'todo completed' : 'todo'}
+    >
+      <label className="todo__status-label">
+        <input
+          data-cy="TodoStatus"
+          type="checkbox"
+          className="todo__status"
+          checked={todo.completed}
+          onChange={handleToggleTodo}
+        />
+      </label>
+
+      {isEditing && (
```

Comment:
```suggestion
      {!!isEditing && (
```
in rendering with && for best rendering control use !! to return a boolean value